cider/type/plan/BinaryExpr.cpp

```diff
@@ -121,8 +122,56 @@ JITExprValue& BinOper::codegenFixedSizeColCmpFun(JITValuePointer& null,
     default:
       UNREACHABLE();
   }
+}
 
-  return expr_var_;
+JITExprValue& BinOper::codegenFixedSizeLogicalFun(JITValuePointer& null,
+                                                  JITValue& lhs,
+                                                  JITValue& rhs) {
+  switch (get_optype()) {
+    case kAND:
+      return set_expr_value(null, lhs && rhs);
+    case kOR:
+      return set_expr_value(null, lhs || rhs);
+    default:
+      UNREACHABLE();
+  }
+}
+
+JITExprValue& BinOper::codegenFixedSizeDistinctFrom(JITFunction& func,
+                                                    FixSizeJITExprValue& lhs_val,
+                                                    FixSizeJITExprValue& rhs_val) {
+  switch (get_optype()) {
+    case kBW_NE: {
+      JITValuePointer value = func.createVariable(JITTypeTag::BOOL, "bw_ne");
+      if (lhs_val.isNullable() && rhs_val.isNullable()) {
+        // both not null and value not equal
+        value = (!lhs_val.getNull() && !rhs_val.getNull() &&
+                 lhs_val.getValue() != rhs_val.getValue()) ||
+                (lhs_val.getNull() != rhs_val.getNull());
+      } else if (lhs_val.isNullable()) {
+        value = lhs_val.getValue() != rhs_val.getValue() || lhs_val.getNull();
+      } else if (rhs_val.isNullable()) {
+        value = lhs_val.getValue() != rhs_val.getValue() || rhs_val.getNull();
+      }
+      return set_expr_value(func.createConstant(JITTypeTag::BOOL, false), value);
+    }
+    case kBW_EQ: {
```

Comment:
Seems this implementation could be simplified. For example, result of ```kBW_EQ``` could be calculated from the ```NOT``` of the result of ```kBW_NE```.

Reply:
The original implementation of `lower_bw_eq` and `lower_bw_ne` has separate logic. But in semantic, we can leverage the `NOT` op. Updated it and will remove the nullable check after [PR](https://github.com/intel/BDTK/pull/218/) merged which should be more straight forward.